### PR TITLE
Anqi performance - add sort by name & grade

### PR DESCRIPTION
**Pull request by Anqi-nus**

### CODE REVIEW

**Comment by Anqi-nus** [comment]
fixed bug when add performance has no parameter

**Comment by benchan911** [comment]
Anqi, please also update the help in UI, if you haven't done so

File: src/main/java/seedu/command/performance/SortPerformanceListByGrade.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        this.performances = performances;
+        this.ui = new UI();
+    }
+
+    /**
+     * Method to sort an performance list according to name.
+     */
+    private void sort() {
+        if (performances.isEmpty()) {
+            ui.displayAttendanceMessage("An empty list cannot be sorted");
```
Comment (benchan911) [inline src/main/java/seedu/command/performance/SortPerformanceListByGrade.java]: Maybe you should change it to ui.displayPerformanceMessage

File: src/main/java/seedu/command/performance/SortPerformanceListByName.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        this.performances = performances;
+        this.ui = new UI();
+    }
+
+    /**
+     * Method to sort an performance list according to name.
+     */
+    private void sort() {
+        if (performances.isEmpty()) {
+            ui.displayAttendanceMessage("An empty list cannot be sorted");
```
Comment (benchan911) [inline src/main/java/seedu/command/performance/SortPerformanceListByName.java]: displayAttendanceMessage to displayPerformanceMessage even though they have the same method of printing

File: src/main/java/seedu/performance/Performance.java [hunk 3]
```diff
@@ -1,5 +1,9 @@
 package seedu.performance;
 
+import seedu.attendance.Attendance;
```
Comment (benchan911) [inline src/main/java/seedu/performance/Performance.java]: Is it an unnecessary import?

File: src/main/java/seedu/command/interpreter/PerformanceCommandInterpreter.java [hunk 4]
```diff
@@ -1,9 +1,7 @@
 package seedu.command.interpreter;
 
 import seedu.command.Command;
-import seedu.command.performance.AddPerformance;
-import seedu.command.performance.DeletePerformance;
-import seedu.command.performance.ListStudentPerformance;
+import seedu.command.performance.*;
```
Comment (benchan911) [inline src/main/java/seedu/command/interpreter/PerformanceCommandInterpreter.java]: Would this fail doing the checkstylemain?

File: src/main/java/seedu/command/interpreter/PerformanceCommandInterpreter.java [hunk 5]
```diff
@@ -37,11 +35,22 @@ public Command decideCommand(String commandDescription) throws DukeException {
             return new DeletePerformance(performances, eventName);
         case "list":
             return new ListStudentPerformance(performances);
+        case "sort":
+            return getSortCommand();
```
Comment (benchan911) [inline src/main/java/seedu/command/interpreter/PerformanceCommandInterpreter.java:43]: this feels inconsistent with the other commands

**Review by benchan911 — CHANGES_REQUESTED** [review]
Please make the respective changes

File: src/main/java/seedu/performance/PerformanceList.java [hunk 6]
```diff
@@ -7,6 +7,9 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import static seedu.performance.Performance.performanceListNameComparator;
+import static seedu.performance.Performance.performanceListGradeComparator;
+
 public class PerformanceList {
     public static ArrayList<Performance> performanceList;
```
Comment (lowjiayee) [inline src/main/java/seedu/performance/PerformanceList.java]: THISSSSS pls change to non-static

Comment (lowjiayee) [reply src/main/java/seedu/performance/PerformanceList.java]: This causes all events to share the same performanceList!!!

File: src/main/java/seedu/command/performance/SortPerformanceListByName.java (hunk 2 above)
Comment (Anqi-nus) [reply src/main/java/seedu/command/performance/SortPerformanceListByName.java]: changed all to ui.displayMessage()

File: src/main/java/seedu/command/interpreter/PerformanceCommandInterpreter.java (hunk 4 above)
Comment (Anqi-nus) [reply src/main/java/seedu/command/interpreter/PerformanceCommandInterpreter.java]: this is an outdated version

**Pull request closed without merging**